packages/runtime/src/useCases/consumption/attributes/GetSharedVersionsOfIdentityAttribute.ts

```diff
@@ -25,29 +27,24 @@ export class GetSharedVersionsOfIdentityAttributeUseCase extends UseCase<GetShar
     }
 
     protected async executeInternal(request: GetSharedVersionsOfIdentityAttributeRequest): Promise<Result<LocalAttributeDTO[]>> {
-        const attribute = await this.attributeController.getLocalAttribute(CoreId.from(request.attributeId));
+        const repositoryAttributeId = CoreId.from(request.attributeId);
+        const repositoryAttribute = await this.attributeController.getLocalAttribute(repositoryAttributeId);
 
-        if (typeof attribute === "undefined") {
-            throw RuntimeErrors.general.recordNotFound(LocalAttribute);
+        if (typeof repositoryAttribute === "undefined") {
+            return Result.fail(RuntimeErrors.general.recordNotFound(LocalAttribute));
         }
 
-        if (attribute.isRelationshipAttribute()) {
-            throw RuntimeErrors.general.invalidPropertyValue("Attribute '${request.attributeId}' is a RelationshipAttribute.");
+        if (!repositoryAttribute.isRepositoryAttribute()) {
+            return Result.fail(RuntimeErrors.attributes.isNoIdentityAttribute(repositoryAttributeId));
```

Comment:
I'm now really confused, but we could fail with isNoRepositoryAttribute now, right? ;)